iis-workflows/iis-workflows-ingest-pmc/src/main/java/eu/dnetlib/iis/workflows/ingest/pmc/metadata/MetadataImporter.java

```diff
@@ -46,15 +85,29 @@ protected void map(AvroKey<DocumentText> key, NullWritable value, Context contex
 				saxParser.parse(new InputSource(
 						new StringReader(nlm.getText().toString())), 
 						pmcXmlHandler);
-                context.write(new AvroKey<ExtractedDocumentMetadata>(output.build()), 
-                		NullWritable.get());
+				mos.write(namedOutputMeta, 
+						new AvroKey<ExtractedDocumentMetadata>(output.build()));
         	 } catch (ParserConfigurationException e) {
-             	throw new IOException("Text extraction failed for id " + nlm.getId() + 
-             			" and text: " + nlm.getText(), e);
+        		 Map<CharSequence, CharSequence> auditSupplementaryData = new HashMap<CharSequence, CharSequence>();
+        			auditSupplementaryData.put(FAULT_TEXT, nlm.getText());
```

Comment:
I think it would be even better to keep this schema-related information directly in our code base. I'm thinking about putting comment on the output port producing `Fault` structure in pmcingestion and metadataextraction modules that says what `supplementaryData` entries are produced. To be more precise, this comment should probably go into `parameters/property/description` field of Oozie XML defining subworkflows corresponding to pmcingestion and metadataextraction modules.

I would also extend the comment describing `supplementaryData` field in the `Fault` structure to something like this: "Other supplementary data related to specific type of fault. See in-code documentation (in Oozie XML) of modules that use this structure for information what exactly they put in here."

What do you think?